File: 121-BestTimetoBuyandSellStock/121-BestTimetoBuyandSellStock.c
int maxProfit(int* prices, int pricesSize) {
    int cheaper=prices[0];
    int max=0;
    int cur=0;
    for(int i=1;i<pricesSize;i++){
        if(prices[i]<cheaper){
            cheaper=prices[i];
        }
        else{
            cur=prices[i]-cheaper;
            if(max<cur){
                max=cur;
            }
        }
    }
    return max;
}

File: 121-BestTimetoBuyandSellStock/121-BestTimetoBuyandSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int cheap=prices[0];
        int max=0;
        for(int i=1;i<prices.length;i++){
            if(prices[i]<cheap){
                cheap=prices[i];
            }
            else{
                int profit=prices[i]-cheap;
                max=Math.max(max,profit);
            }
        }
        return max;
    }
}